tests/test_dataset.py

```diff
@@ -61,6 +61,18 @@ def test_datasets_override_default(dataset_name):
     assert dataset.CLASSES == original_classes
 
 
+@patch.multiple(BaseDataset, __abstractmethods__=set())
+def test_dataset_evaluation():
+    dataset = BaseDataset(data_prefix='', pipeline=[], test_mode=True)
+    dataset.data_infos = [dict(gt_label=0), dict(gt_label=1)]
+    fake_results = np.array([[1, 0], [0, 1]])
```

Comment:
how about more complex test samples?
For example, use 3 classes, and with repeated results?